### PR TITLE
Add DataCrunch provider

### LINKED ISSUE #8: DataCrunch
Hi!

Datacrunch.io is an interesting modern ML cloud with premium dedicated GPU servers and clusters.
I'll add support for this provider.

### PR DESCRIPTION
**Pull request by TheBits**
Close #8

### CODE REVIEW

File: .github/workflows/catalogs.yml [hunk 1]
```diff
@@ -63,6 +63,31 @@ jobs:
           path: azure.csv
           retention-days: 1
 
+  catalog-nebius:
```
Comment (Egor-S) [inline .github/workflows/catalogs.yml:66]: Leftovers from copypaste, I guess catalog-datacrunch was supposed

File: src/gpuhunt/providers/datacrunch.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    def get(self, query_filter: Optional[QueryFilter] = None) -> List[RawCatalogItem]:
+        items = [
+            RawCatalogItem(
+                instance_name=instance.instance_type,
+                location="FIN-01",
+                price=instance.price_per_hour,
+                cpu=instance.cpu["number_of_cores"],
+                memory=instance.memory["size_in_gigabytes"],
+                gpu_count=instance.gpu["number_of_gpus"],
+                gpu_name=instance.gpu["description"],
```
Comment (Egor-S) [inline src/gpuhunt/providers/datacrunch.py:24]: Is this description normalized to our format? A100, A10, RTX4090?

**Review by Egor-S — COMMENTED** [review]
Does the list of DataCrunch instances stay the same despite current availability?

If yes, I suggest adding a couple of integrity tests before merging.
If not, I suggest adding this provider to ONLINE_PROVIDERS

**Comment by Egor-S** [comment]
Please, add a couple of catalog tests to `src/integrity_tests/test_datacrunch.py`

File: .github/workflows/catalogs.yml (hunk 1 above)
Comment (TheBits) [reply .github/workflows/catalogs.yml]: Yes this was copypaste. I fixed it.

**Comment by TheBits** [comment]
Thanks Egor for your review!
I 've added integrity_tests.

File: src/gpuhunt/_internal/models.py [hunk 3]
```diff
@@ -4,6 +4,12 @@
 from gpuhunt._internal.utils import empty_as_none
 
 
+def bool_loader(x: Union[bool, str]) -> bool:
+    if type(x) == bool:
```
Comment (Egor-S) [inline src/gpuhunt/_internal/models.py:8]: Shouldn't we use `isinstance(x, bool)` here?

**Review by Egor-S — COMMENTED** [review]
LGTM

**Pull request merged**